snowshu/adapters/target_adapters/postgres_adapter/postgres_adapter.py

```diff
@@ -113,6 +115,10 @@ def create_database_if_not_exists(self, database: str) -> str:
         return database
 
     def create_schema_if_not_exists(self, database: str, schema: str) -> None:
+        database = database if self.preserve_case \
+            else correct_case(database, self.DEFAULT_CASE == 'upper')
+        schema = schema if self.preserve_case \
+            else correct_case(schema, self.DEFAULT_CASE == 'upper')
```

Comment:
There's already a function to encapsulation this behavior in `BaseSQLAdapter`, `_correct_case`. If we need to check the `preserve_case` flag or make any sort of case correction, we can just use that function.